test.cpp

```diff
@@ -3299,11 +3299,44 @@ static void preprocess_files()
     }
 }
 
-static void safe_api()
+static void tokenlist_api()
 {
+    std::vector<std::string> filenames;
+    {
+        char input[] = "code"; // NOLINT(misc-const-correctness)
```

Comment:
This is not actually reported but per-emptively added in case it is detected in the future. I added a comment about it upstream: https://github.com/llvm/llvm-project/issues/56758#issuecomment-3638935265